lightningd/bitcoind.c

```diff
@@ -102,98 +102,75 @@ static void bitcoin_plugin_error(struct bitcoind *bitcoind, const char *buf,
 	      toks->end - toks->start, buf + toks->start);
 }
 
-/* `getfeerate`
+/* `estimatefees`
  *
  * Gather feerate from our Bitcoin backend. Will set the feerate to `null`
  * if estimation failed.
  *
  * Plugin response:
  * {
- *	"feerate": <fee per KB>,
+ *	"urgent": <btc per KB>,
+ *	"normal": <btc per KB>,
+ *	"slow": <btc per KB>
  * }
  */
 
 struct estimatefee_call {
 	struct bitcoind *bitcoind;
-	size_t i;
-	const u32 *blocks;
-	const char **estmode;
-
 	void (*cb)(struct bitcoind *bitcoind, const u32 satoshi_per_kw[],
 		   void *);
 	void *arg;
-	u32 *satoshi_per_kw;
 };
 
-static void do_one_estimatefee(struct bitcoind *bitcoind,
-			       struct estimatefee_call *call);
-
-static void getfeerate_callback(const char *buf, const jsmntok_t *toks,
-				const jsmntok_t *idtok,
-				struct estimatefee_call *call)
+static void estimatefees_callback(const char *buf, const jsmntok_t *toks,
+				  const jsmntok_t *idtok,
+				  struct estimatefee_call *call)
 {
 	const jsmntok_t *resulttok, *feeratetok;
-	u64 feerate;
+	const char *targets[] = {"urgent", "normal", "slow"};
+	u32 *feerates = tal_arr(call, u32, 3);
 
 	resulttok = json_get_member(buf, toks, "result");
 	if (!resulttok)
 		bitcoin_plugin_error(call->bitcoind, buf, toks,
-				     "getfeerate",
+				     "estimatefees",
 				     "bad 'result' field");
 
-	feeratetok = json_get_member(buf, resulttok, "feerate");
-	if (!feeratetok)
-		bitcoin_plugin_error(call->bitcoind, buf, toks,
-				     "getfeerate",
-				     "bad 'feerate' field");
+	for (size_t i = 0; i < tal_count(feerates); i++) {
+		feeratetok = json_get_member(buf, resulttok, targets[i]);
+		if (!feeratetok)
+			bitcoin_plugin_error(call->bitcoind, buf, toks,
+					     "estimatefees",
+					     "bad 'urgent', 'normal' or 'slow'"
+					     " field");
 
-	/* FIXME: We could trawl recent blocks for median fee... */
-	if (!json_to_u64(buf, feeratetok, &feerate)) {
-		log_unusual(call->bitcoind->log, "Unable to estimate %s/%u fee",
-			    call->estmode[call->i], call->blocks[call->i]);
+		if (!json_to_u32(buf, feeratetok, &feerates[i])) {
+			log_unusual(call->bitcoind->log,
+				    "Unable to estimate %s fees",
+				    targets[i]);
 
 #if DEVELOPER
-		/* This is needed to test for failed feerate estimates
-		 * in DEVELOPER mode */
-		call->satoshi_per_kw[call->i] = 0;
+			/* This is needed to test for failed feerate estimates
+			* in DEVELOPER mode */
+			feerates[i] = 0;
```

Comment:
the better way to do this is to use a config-flag to trigger this behavior if you need it for tests. 

actually the better option would be to use a config flag for the floor for all cases, which the test can then set to zero if desired. (which removes the switch for `testnet` below, also

Reply:
I did not modify this